src/controllers/books.controller.js

```diff
@@ -0,0 +1,103 @@
+const { isObjEmpty } = require('../utils');
+const model = require('../models/books.model');
+
+const getAllBooks = async (req, res) => {
+  try {
+    const books = await model.getAllBooks(req);
+
+    res.status(200).json({
+      books,
+    });
+  } catch (error) {
+    console.error('[ERROR]', error);
+    res.sendStatus(500);
+  }
+};
+
+const getBookById = async (req, res) => {
+  const bookId = req.params.id;
+  try {
+    const book = await model.getBookById(bookId);
+
+    isObjEmpty(book)
+      ? res
+          .status(404)
+          .json({ error: 'A book with the provided ID does not exist' })
+      : res.status(200).json({
+          book,
+        });
+  } catch (error) {
+    console.error('[ERROR]', error);
+    res.sendStatus(500);
+  }
+};
+
+const postBook = async (req, res) => {
+  const params = Object.values(req.body);
+
+  if (params.length < 6) {
+    res.status(400).json({ error: 'Missing fields in the request body' });
+    return;
+  }
+
+  try {
+    const book = await model.postBook(params);
+
+    res.status(201).json({
+      book,
+    });
+  } catch (error) {
+    console.error('[ERROR]', error);
+    res.sendStatus(500);
+  }
+};
+
+const updateBook = async (req, res) => {
+  const { title } = req.body;
+
+  const bookId = req.params.id;
+  const params = Object.values(req.body);
+
+  try {
+    const book = await model.updateBook(params, bookId, title);
+
+    if (book === 'title already exists') {
```

Comment:
Nice. I'd recommend moving strings like this into their own constant somewhere so you can avoid having the same string scattered throughout your app, which can cause issues if you ever make a typo.

`if (book === TITLE_EXISTS)` is much easier to maintain